### DISCUSSION TITLE
sbt-tpolecat vs sbt-typelevel-settings

### DISCUSSION THREAD

**Discussion by BennyMcBenBen** [DavidGregory084/sbt-tpolecat#109] [General]
Hello. Love this plugin. I discovered sbt-typelevel-settings also exists and I am confused. Will this plugin continue to be developed or is sbt-typelevel-settings going to replace this? They work a bit differently so I am curious.

**Comment by armanbilge** [comment]
Good question! Currently the plan is for both to be developed; sbt-typelevel-settings is not intended as a replacement by any means.

> They work a bit differently

That's currently the case, although they will hopefully be more similar in the next version of sbt-typelevel/sbt-tpolecat thanks to the scalac-options library.
https://github.com/typelevel/scalac-options

With that said, the main difference is that sbt-typelevel-settings adds some additional configuration besides `scalacOptions`, for example:
- the `kind-projector` and `better-monadic-for` compiler plugins
- scaladoc configuration
- additional source directories e.g. `scala-2.12-`, `scala-2.13+`
- JDK release version and API mappings
- `javacOptions`

Many of these features are more useful for (OSS) library development, rather than applications, since that is the primary user base of sbt-typelevel.

**Reply by BennyMcBenBen** [reply]
Thanks! This clears things up

**Comment by DavidGregory084** [comment]
Great question - @armanbilge and I have been coordinating on how to merge the two plugins. At the moment the plan is to move both plugins to use a shared dependency which implements the core logic of this plugin - see #107.

**Reply by DavidGregory084** [reply]
Haha Arman you are so speedy. What @armanbilge said basically 😆
